client/src/modules/stock/entry/modals/findDonation.modal.js

```diff
@@ -0,0 +1,117 @@
+angular.module('bhima.controllers')
+  .controller('StockFindDonationModalController', StockFindDonationModalController);
+
+StockFindDonationModalController.$inject = [
+  '$uibModalInstance', 'DonationService', 'NotifyService',
+  'uiGridConstants', 'GridFilteringService', 'ReceiptModal',
+  'bhConstants',
+];
+
+function StockFindDonationModalController(
+  Instance, Donation, Notify,
+  uiGridConstants, Filtering, Receipts, bhConstants,
+) {
+  const vm = this;
+
+  // global
+  vm.selectedRow = {};
+
+  /* ======================= Grid configurations ============================ */
+  vm.filterEnabled = false;
+  vm.gridOptions = { appScopeProvider : vm };
+
+  const filtering = new Filtering(vm.gridOptions);
+
+  const columns = [
+    {
+      field            : 'reference',
+      displayName      : 'TABLE.COLUMNS.REFERENCE',
+      headerCellFilter : 'translate',
+      cellTemplate     : 'modules/stock/entry/modals/templates/purchase_reference.tmpl.html',
```

Comment:
It seems like this template is used also with purchase, do you think we can rename it ?